test/test_viscous.py

```diff
@@ -502,45 +503,51 @@ def test_viscous_timestep(actx_factory, dim, mu, vel):
     )
 
     order = 1
+    gamma = 1.4
+    if mu < 0:
+        mu = 0
+        tv_model = None
+    else:
+        tv_model = SimpleTransport(viscosity=mu)
 
     discr = EagerDGDiscretization(actx, mesh, order=order)
     zeros = discr.zeros(actx)
     ones = zeros + 1.0
 
-    velocity = make_obj_array([zeros+vel for _ in range(dim)])
+    vel_mag = np.sqrt(dim*vel_val**2)
+    pressure = mass_val / gamma
+    ke_val = mass_val * vel_mag * vel_mag/2.
+    ie_val = pressure/(gamma - 1)
+    energy_total_val = ie_val + ke_val
+    wavespeed_val = vel_mag + 1.0
+    nu = mu / mass_val
 
-    massval = 1
-    mass = massval*ones
+    # we assume "chlen" in these tests is correct
+    from grudge.dt_utils import characteristic_lengthscales
+    chlen = characteristic_lengthscales(actx, discr)
+    # from grudge.op import nodal_min
+    # chlen_min = nodal_min(discr, "vol", chlen)
 
-    # I *think* this energy should yield c=1.0
-    energy = zeros + 1.0 / (1.4*.4)
+    dt_expected = chlen / (wavespeed_val + (nu / chlen))
+    print(f"{dt_expected=}")
+    tol = 1e-12
+
+    mass = mass_val*ones
+    energy = zeros + energy_total_val
+    velocity = make_obj_array([zeros+vel_val for _ in range(dim)])
     mom = mass * velocity
     species_mass = None
 
     cv = make_conserved(dim, mass=mass, energy=energy, momentum=mom,
                         species_mass=species_mass)
 
-    from grudge.dt_utils import characteristic_lengthscales
-    chlen = characteristic_lengthscales(actx, discr)
-    from grudge.op import nodal_min
-    chlen_min = nodal_min(discr, "vol", chlen)
-
-    mu = mu*chlen_min
-    if mu < 0:
-        mu = 0
-        tv_model = None
-    else:
-        tv_model = SimpleTransport(viscosity=mu)
-
     eos = IdealSingleGas()
     gas_model = GasModel(eos=eos, transport=tv_model)
     fluid_state = make_fluid_state(cv, gas_model)
 
     from mirgecom.viscous import get_viscous_timestep
     dt_field = get_viscous_timestep(discr, fluid_state)
+    print(f"{dt_field=}")
 
-    speed_total = fluid_state.wavespeed
-    dt_expected = chlen / (speed_total + (mu / chlen))
-
-    error = (dt_expected - dt_field) / dt_expected
-    assert actx.to_numpy(discr.norm(error, np.inf)) == 0
+    error = (dt_field - dt_expected) / dt_expected
+    assert actx.to_numpy(discr.norm(error, np.inf)) < tol
```

Comment:
IIUC this doesn't actually check the stability of the PDE solver at these time steps. Is that something we could incorporate?

Reply:
That's right, it only checks that we implemented the formula correctly.   I think we decided at today's meetings that the validation test(s) for the solver do not need to be integrated and we can add a documentation blurb about (a) reproducing the tests using the provided examples, and (b) how to estimate stability limits for your given problem.

Reply:
> (...) Is that something we could incorporate?

I imagine we can test for a contrived case at what CFL it goes unstable (i.e. when infs, or nans appear in the derived quantitites), and make sure that is somewhere between .1 - 1.1 , or even tighter for inviscid problems.  This is how I've checked it by-hand.

To do this in the unit testing, we need a NS and Euler test steppers that support constant CFL timestepping.  One for Euler is here:
https://github.com/illinois-ceesd/mirgecom/blob/9b19b749de2c7d099f4cd5aee67727eda8349225/test/test_euler.py#L506,

but I do not believe I have tested it with constant CFL yet.   An NS stepper is not something that can go into this PR as NS is downstream of this.

I wonder if it is worth it to test it at this level?  Our timestep estimator is designed to return a conservative estimate for the limiting simulation-wide DT, or CFL, so I think as long as it gives a ballpark measure of the {DT, CFL}, i.e. if the simulation is "stable" at CFL < [.1, 1.1] , then it's ok for this purpose.    

A useful thing to have would be a routine that returns precisely (up to `characterstic_length_scales`) the limiting timestep for each of advection, thermal diffusion, species diffusion, and chemistry, at least so they can be visualized.  Taking the min DT among these should ensure stability.  What we have currently globs them all together to get a conservative value that should ensure stability for the current state.

Reply:
https://github.com/inducer/grudge/pull/229